### PR TITLE
New Docker Stack Task Info Module with Tests

### LINKED ISSUE #729: New Docker Stack Task Info Module
##### SUMMARY

I would like to list all stacks tasks from a given stack name like `docker stack ps` in order to run some setup containers if a particular stack is not present.

##### ISSUE TYPE

- Feature Idea

##### COMPONENT NAME

docker_stack_task_info.py

##### ADDITIONAL INFORMATION

```yaml
   - name: List all docker stacks
     docker_stack_task_info:
       name: test_stack      # return list tasks from specific stack name
```

### PR DESCRIPTION
**Pull request by imjoseangel**
##### SUMMARY

Implement docker_stack_task_info functionality

Fixes #729

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

`docker_stack_task_info.py`

##### ADDITIONAL INFORMATION

Shows Docker stack tasks from a given stack name

```yaml
---
- name: Docker stack list
  hosts: localhost
  connection: local
  gather_facts: false

  tasks:

    - name: List stack
      docker_stack_info:
      register: output

    - name: show stack
      debug:
        msg: "{{ output.results }}"

    - name: Get stack tasks from first element
      docker_stack_task_info:
        name: "{{ output.results[0].Name }}"
      register: output

    - name: Show stack tasks
      debug:
        msg: "{{ output.results }}"
```

Output

```json
ok: [localhost] => {
    "msg": [
        {
            "current state": "Running 24 hours ago",
            "desired state": "Running",
            "error": "",
            "id": "eda0cda3-1bc",
            "image": "grafana/grafana",
            "name": "grafana_grafana-0",
            "node": "docker-desktop",
            "ports": "*:0->3000/tcp"
        },
        {
            "current state": "Running 24 hours ago",
            "desired state": "Running",
            "error": "",
            "id": "075e8ac3-d28",
            "image": "graphiteapp/graphite-statsd",
            "name": "grafana_graphite-5b5b99dd5f-sk4rh",
            "node": "docker-desktop",
            "ports": "*:0->80/tcp,*:0->2003/tcp,*:0->2004/tcp"
        }
    ]
}
```

### CODE REVIEW

File: plugins/modules/docker_stack_tasks_info.py [hunk 1]
```diff
@@ -0,0 +1 @@
+plugins/modules/cloud/docker/docker_stack_tasks_info.py
```
Comment (felixfontein) [inline plugins/modules/docker_stack_tasks_info.py:1]:
```suggestion
cloud/docker/docker_stack_tasks_info.py
```

File: plugins/modules/cloud/docker/docker_stack_task_info.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+RETURN = '''
+results:
+    description: |
+        List of dictionaries containing the list of tasks associated
+        to a stack name.
+    sample: >
+        "results": [{"CurrentState":"Running 9 minutes ago","DesiredState":"Running","Error":"","ID":"7wqv6m02ugkw","Image":"busybox:latest","Name":"test_stack_busybox.1","Node":"docker-desktop","Ports":""}]
+    returned: always
+    type: list
```
Comment (felixfontein) [inline plugins/modules/cloud/docker/docker_stack_task_info.py:30]:
```suggestion
    type: list
    elements: dict
```

File: plugins/modules/cloud/docker/docker_stack_task_info.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+DOCUMENTATION = '''
+---
+module: docker_stack_task_info
+author: "Jose Angel Munoz (@imjoseangel)"
+short_description: Return information of the tasks on a docker stack
+description:
+  - Retrieve information on docker stacks tasks using the C(docker stack) command
+    on the target node (see examples).
+version_added: "1.1.0"
```
Comment (felixfontein) [inline plugins/modules/cloud/docker/docker_stack_task_info.py:19]:
```suggestion
version_added: "1.1.0"
options:
  name:
    ...
```

File: plugins/modules/cloud/docker/docker_stack_task_info.py (hunk 2 above)
Comment (imjoseangel) [reply plugins/modules/cloud/docker/docker_stack_task_info.py:36]: Thanks for your comments @felixfontein

File: plugins/modules/cloud/docker/docker_stack_task_info.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+module: docker_stack_task_info
+author: "Jose Angel Munoz (@imjoseangel)"
+short_description: Return information of the tasks on a docker stack
+description:
+  - Retrieve information on docker stacks tasks using the C(docker stack) command
+    on the target node (see examples).
+options:
+  name:
+    description:
+      - Stack name
```
Comment (felixfontein) [inline plugins/modules/cloud/docker/docker_stack_task_info.py:22]:
```suggestion
      - Stack name.
```

File: plugins/modules/cloud/docker/docker_stack_task_info.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+version_added: "1.1.0"
+'''
+
+RETURN = '''
+results:
+    description: |
+        List of dictionaries containing the list of tasks associated
+        to a stack name.
+    sample: >
+        "results": [{"CurrentState":"Running","DesiredState":"Running","Error":"","ID":"7wqv6m02ugkw","Image":"img","Name":"img.1","Node":"swarm","Ports":""}]
```
Comment (felixfontein) [inline plugins/modules/cloud/docker/docker_stack_task_info.py:34]:
```suggestion
        [{"CurrentState":"Running","DesiredState":"Running","Error":"","ID":"7wqv6m02ugkw","Image":"img","Name":"img.1","Node":"swarm","Ports":""}]
```

**Comment by felixfontein** [comment]
@imjoseangel do you want to continue with this, or should I merge?

**Comment by imjoseangel** [comment]
Job is done. Thank you @felixfontein , you can merge.

**Comment by felixfontein** [comment]
@imjoseangel thanks a lot for contributing this!

**Pull request merged**